src/adaptors/yjs-ndn-adaptor.ts

```diff
@@ -96,9 +115,148 @@ export class NdnSvsAdaptor {
+    if (count % 5 == 0) {
+      console.log("It's time to make a snapshot!");
+      console.log('debug: group prefix: ', this.syncAgent.appPrefix.toString());
+      const encodedSV = Encoder.encode(stateVector);
+
+      // NOTE: The following code depend on snapshot naming convention to work.
+      // Verify this part if there's a change in naming convention.
+      // TODO: Currently naming convention is hard-coded. May need organizing.
+      const snapshotPrefix = this.syncAgent.appPrefix.append('32=snapshot');
+      // New SVS encodings
+      const snapshotName = snapshotPrefix.append(new Component(Version.type, encodedSV));
+      console.log('debug: targeted snapshot Prefix (persistStore key): ', snapshotPrefix.toString());
+      // /groupPrefix/32=snapshot/
+      console.log('debug: targeted snapshot Name: ', snapshotName.toString());
+      // /groupPrefix/32=snapshot/54=<stateVector>
+      const decodedSV = Decoder.decode(snapshotName.at(-1).value, StateVector);
+      console.log('debug: decoding encoded SV from snapshotName: ', decodedSV);
+      let count = 0;
+      for (const [_id, seq] of decodedSV) {
+        count += seq;
+      }
+      console.log('debug: decoding encoded SV total packet count: ', count);
+      console.log('This should match the state vector in the debug page and the previous count before encoding');
+
+      // Snapshot content generation
+      const content = Y.encodeStateAsUpdate(this.doc);
+      // its already in UInt8Array (binary), transporting currently without any additional encoding.
+      console.log('yjs backend data: ', content);
+
+      // use syncAgent's blob and publish mechanism
+      await this.syncAgent.publishBlob('snapshot', content, snapshotName, true);
+
+      // NOTE: The following code depend on snapshot naming convention to work.
+      // Verify this part if there's a change in naming convention.
+      // Race Condition note: Testing suggests that the write above with publishBlob()
+      //                      is near certainly done before the read happens below.
+      //                      Hence no delay is added.
+      // first segmented object is at /50=%00
+      const firstSegmentName = snapshotName.append('50=%00').toString();
+      console.log('debugTargetName: ', firstSegmentName);
+      const firstSegmentPacketEncoded = await this.syncAgent.persistStorage.get(firstSegmentName);
+      if (firstSegmentPacketEncoded) {
+        const firstSegmentPacket = Decoder.decode(firstSegmentPacketEncoded, Data);
+        console.log('persistentStore check: ', firstSegmentPacket);
+        console.log('persistentStore check Data Name:', firstSegmentPacket.name.toString());
+        await this.syncAgent.persistStorage.set(snapshotPrefix.toString(), Encoder.encode(firstSegmentPacket));
+      }
+    }
+  }
+  // End Injection point 1
+
+  // -- Adam Chen Injection Point 3: HandleSnapshotUpdate --
+  async handleSnapshotUpdate(snapshotName: Uint8Array) {
```

Comment:
This is wrong signature. Publish and listener are paired.
On L171, you published the encoded state as content.
Therefore, the first argument of this listener will be the content, not the name.